x-pack/legacy/plugins/ml/public/data_visualizer/components/field_data_card/content_types/number_content.tsx

```diff
@@ -0,0 +1,200 @@
+  const [distributionChartData, setDistributionChartData] = useState(defaultChartData);
+
+  const detailsOptions = [
+    {
+      value: DETAILS_MODE.DISTRIBUTION,
+      text: i18n.translate('xpack.ml.fieldDataCard.cardNumber.details.distributionOfValuesLabel', {
+        defaultMessage: 'distribution of values',
+      }),
+    },
+    {
+      value: DETAILS_MODE.TOP_VALUES,
+      text: i18n.translate('xpack.ml.fieldDataCard.cardNumber.details.topValuesLabel', {
+        defaultMessage: 'top values',
+      }),
+    },
+  ];
+
+  return (
+    <div className="mlFieldDataCard__stats">
+      <div>
+        <EuiIcon type="document" />
+        &nbsp;
+        <FormattedMessage
+          id="xpack.ml.fieldDataCard.cardNumber.documentsCountDescription"
+          defaultMessage="{count, plural, zero {# document} one {# document} other {# documents}} ({docsPercent}%)"
+          values={{
+            count,
+            docsPercent,
+          }}
+        />
+      </div>
+
+      <EuiSpacer size="xs" />
+
+      <div>
+        <EuiIcon type="database" />
+        &nbsp;
+        <FormattedMessage
+          id="xpack.ml.fieldDataCard.cardNumber.distinctCountDescription"
+          defaultMessage="{cardinality} distinct {cardinality, plural, zero {value} one {value} other {values}}"
+          values={{
+            cardinality,
+          }}
+        />
+      </div>
+
+      <EuiSpacer size="xs" />
+
+      <EuiFlexGroup gutterSize="xs" justifyContent="center">
+        <EuiFlexItem grow={1}>
+          <FormattedMessage id="xpack.ml.fieldDataCard.cardNumber.minLabel" defaultMessage="min" />
+        </EuiFlexItem>
+        <EuiFlexItem grow={1}>
+          <FormattedMessage
+            id="xpack.ml.fieldDataCard.cardNumber.medianLabel"
+            defaultMessage="median"
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={1}>
+          <FormattedMessage id="xpack.ml.fieldDataCard.cardNumber.maxLabel" defaultMessage="max" />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiFlexGroup gutterSize="xs" justifyContent="center">
+        <EuiFlexItem grow={1} className="eui-textTruncate">
+          <DisplayValue value={kibanaFieldFormat(min, fieldFormat)} />
+        </EuiFlexItem>
+        <EuiFlexItem grow={1} className="eui-textTruncate">
+          <DisplayValue value={kibanaFieldFormat(median, fieldFormat)} />
+        </EuiFlexItem>
+        <EuiFlexItem grow={1} className="eui-textTruncate">
+          <DisplayValue value={kibanaFieldFormat(max, fieldFormat)} />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiSpacer size="s" />
+
+      <EuiFlexGroup gutterSize="xs" justifyContent="center">
+        <EuiFlexItem grow={false} style={{ width: 200 }} className="eui-textTruncate">
+          <EuiSelect
+            options={detailsOptions}
+            value={detailsMode}
+            compressed
+            onChange={e => setDetailsMode(e.target.value as DETAILS_MODE)}
```

Comment:
This might be better as a separate function defined somewhere above so that it doesn't get recreated on every render. 🤔 I don't _think_ you'd need to explicitly pass the event to it if you just had something like `handleOnChange` defined somewhere.

Reply:
Looking at this [React Hooks FAQ](https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render), I'm inclined to leave this inline definition as it is, with the performance overhead unlikely to be significant, and prefer its improved readability.

Reply:
👍